atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetterTest.java

```diff
@@ -0,0 +1,178 @@
+        when(tableNameUnmapper.getLongTableNamesFromMappingTable(connectionSupplier, SHORT_TABLE_NAMES))
+                .thenReturn(getLongOverflowTableNames());
+
+        Set<TableReference> tableReferences = tableMappingTableNameGetter.getTableReferencesFromShortOverflowTableNames(
+                connectionSupplier, SHORT_TABLE_NAMES);
+        assertThat(tableReferences).isEqualTo(TABLE_REFERENCES);
+    }
+
+    @Test
+    public void getTableReferencesFromShortTableNamesTransformsProvidedNamesWhenMappingDisabled()
+            throws TableMappingNotFoundException {
+        Set<TableReference> tableReferences = nonTableMappingTableNameGetter.getTableReferencesFromShortTableNames(
+                connectionSupplier, getLongTableNames());
+        assertThat(tableReferences).isEqualTo(TABLE_REFERENCES);
+    }
+
+    @Test
+    public void getTableReferencesFromShortOverflowTableNamesTransformsProvidedNamesWhenMappingDisabled()
+            throws TableMappingNotFoundException {
+        Set<TableReference> tableReferences =
+                nonTableMappingTableNameGetter.getTableReferencesFromShortOverflowTableNames(
+                        connectionSupplier, getLongOverflowTableNames());
+        assertThat(tableReferences).isEqualTo(TABLE_REFERENCES);
+    }
+
+    @Test
+    public void getTableReferencesFromShortTableNamesThrowsIfMappingDoesNotExist()
+            throws TableMappingNotFoundException {
+        when(tableNameUnmapper.getLongTableNamesFromMappingTable(connectionSupplier, SHORT_TABLE_NAMES))
+                .thenThrow(TableMappingNotFoundException.class);
+        assertThatThrownBy(() -> tableMappingTableNameGetter.getTableReferencesFromShortTableNames(
+                        connectionSupplier, SHORT_TABLE_NAMES))
+                .isInstanceOf(TableMappingNotFoundException.class);
+    }
+
+    @Test
+    public void getTableReferencesFromShortOverflowTableNamesThrowsIfMappingDoesNotExist()
+            throws TableMappingNotFoundException {
+        when(tableNameUnmapper.getLongTableNamesFromMappingTable(connectionSupplier, SHORT_TABLE_NAMES))
+                .thenThrow(TableMappingNotFoundException.class);
+        assertThatThrownBy(() -> tableMappingTableNameGetter.getTableReferencesFromShortOverflowTableNames(
+                        connectionSupplier, SHORT_TABLE_NAMES))
+                .isInstanceOf(TableMappingNotFoundException.class);
+    }
+
+    @Test
+    public void getTableReferencesFromShortTableNamesThrowsIfMappedLongNameDoesNotBeginWithPrefix()
+            throws TableMappingNotFoundException {
+        when(tableNameUnmapper.getLongTableNamesFromMappingTable(connectionSupplier, SHORT_TABLE_NAMES))
+                .thenReturn(SHORT_TABLE_NAMES);
+        assertThatThrownBy(() -> tableMappingTableNameGetter.getTableReferencesFromShortTableNames(
+                        connectionSupplier, SHORT_TABLE_NAMES))
+                .isInstanceOf(SafeIllegalArgumentException.class);
```

Comment:
Ah, that's why we do them. I was under the impression we did them to ensure that we don't change exception messages willynilly for a consistent API e.t.c rather than to assert the exception comes from the expected source.